### PR TITLE
GVS BigQuery Traversal Artifact [VS-927]

### PR DESCRIPTION
**Pull request by mcovarr**

### CODE REVIEW

File: src/main/java/org/broadinstitute/hellbender/tools/gvs/extract/ExtractCohortToVcf.java [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+import org.broadinstitute.barclay.argparser.CommandLineProgramProperties;
+import org.broadinstitute.barclay.help.DocumentedFeature;
+import org.broadinstitute.hellbender.cmdline.StandardArgumentDefinitions;
+import org.broadinstitute.hellbender.cmdline.programgroups.ShortVariantDiscoveryProgramGroup;
+import org.broadinstitute.hellbender.utils.io.IOUtils;
+
+
+@SuppressWarnings("unused")
+@CommandLineProgramProperties(
+        summary = "(\"ExtractCohortToVcf\") - Filter and extract variants from BigQuery to a VCF output file.",
```
Comment (gbggrant) [inline src/main/java/org/broadinstitute/hellbender/tools/gvs/extract/ExtractCohortToVcf.java:15]: nit: 'VCF output' ? (a VCF is a file)

**Review by gbggrant — APPROVED** [review]
Looks good.

File: src/main/java/org/broadinstitute/hellbender/tools/gvs/common/ExtractTool.java [hunk 2]
```diff
@@ -18,35 +16,23 @@
...
             optional = true
     )
     protected String projectID = null;
 
-
     @Argument(
             fullName = "dataset-id",
             doc = "ID of the Google Cloud dataset to use when executing queries",
-            optional = true // I guess, but wont it break otherwise or require that a dataset be created with the name temp_tables?
+            optional = true // I guess, but won't it break otherwise or require that a dataset be created with the name temp_tables?
```
Comment (RoriCremer) [inline src/main/java/org/broadinstitute/hellbender/tools/gvs/common/ExtractTool.java:32]: truly why do we still allow this parameter at all?

File: src/main/java/org/broadinstitute/hellbender/tools/gvs/extract/ExtractCohort.java [hunk 3]
```diff
@@ -481,16 +473,10 @@ public Object onTraversalSuccess() {
...
+        if (engine != null) {
             logger.info(String.format("***Processed %d total sites", engine.getTotalNumberOfSites()));
             logger.info(String.format("***Processed %d total variants", engine.getTotalNumberOfVariants()));
             logger.info(String.format("***Read API scanned %d bytes", engine.getTotalEstimatedBytesScanned()));
         }
-
-        // Close up our writer if we have to:
-        if ( vcfWriter != null ) {
-            vcfWriter.close();
-        }
```
Comment (RoriCremer) [inline src/main/java/org/broadinstitute/hellbender/tools/gvs/extract/ExtractCohort.java:493]: why dont we need to close this any more?

**Review by RoriCremer — APPROVED** [review]

File: src/main/java/org/broadinstitute/hellbender/tools/gvs/extract/ExtractCohort.java (hunk 3 above)
Comment (mcovarr) [reply src/main/java/org/broadinstitute/hellbender/tools/gvs/extract/ExtractCohort.java:493]: It's closed in the VCF writer subclass that actually does the writing now.

**Comment by mcovarr** [comment]
Hi @cmnbroad, I'll take a look at that and let you know how it goes.

**Comment by cmnbroad** [comment]
@mcovarr Great - thanks for doing that. The number that we need is one that corresponds to the number of variant contexts that will be passed to `apply`. Is that what this does ? 

If so, is there any reason that @KevinCLydon can't/shouldn't start using this branch for his testing ?

**Comment by mcovarr** [comment]
Hi @cmnbroad, no unfortunately that's not what that count represents. 

Per team discussion I'm going to remove that counting code from this branch and merge to our `ah_var_store` branch. I'll reach out on Slack about setting up some time to discuss the issues here.

**Pull request merged**